echopop/survey.py

```diff
@@ -1230,3 +1246,238 @@ def krige( self ,
+
+        ### Merge with the kriged results 
+        kriged_results = (
+            kriged_results
+            .merge( average_weight_df , on = [ 'stratum_num' ] )
+            .assign( N_adult_kriged = lambda df: df.B_adult_kriged / df.average_weight_total )
+            .merge( self.acoustics[ 'sigma_bs' ][ 'strata_mean' ] , on = [ 'stratum_num' ] )
+            .assign( NASC_kriged = lambda df: df.N_adult_kriged * df.sigma_bs_mean )
+        )
+
+        ### Calculate grid CV 
+        biomass_density_adult = (
+            self.biology[ 'population' ][ 'areal_density' ][ 'biomass_density_df' ]
+            .loc[ lambda df: df.sex == 'all' ]
+            .drop_duplicates( subset = [ 'transect_num' , 'latitude' , 'longitude' , 'stratum_num' ] )
+        )
+
+        ### ---------------------------------------------------------------
+        ### !!! TODO: 
+        ### While `C0` produces a similar standard deviation as the original
+        ### code, `Bn` does not. This discrepancy results in an order of 
+        ### magnitude change in the resulting `biomass_adult_cell_CV` estimate.
+        ### The source of this error stems from `kriged_results.B_a_adult_mean`
+        ### whereby the right-tail of the `B_a_adult_mean` distribution is 
+        ### substantially shorter than from values (`biomass_density_adult_mean`)
+        ### produced in the original code. This therefore relates to Issue #202.
+        ### ---------------------------------------------------------------
+        C0 = np.std( biomass_density_adult.B_a , ddof = 1 ) ** 2
+        Bn = np.nansum( kriged_results.B_a_adult_mean * kriged_results.cell_area_nmi2 ) * 1e-9
+        kriged_results = (
+            kriged_results
+            .assign( biomass_adult_cell_CV = lambda df: (
+                self.statistics[ 'kriging' ][ 'model_config' ][ 'A0' ] *
+                np.sqrt( kriged_results.B_a_adult_prediction_variance * C0 ) *
+                1e-9 /
+                Bn *
+                np.sqrt( len( kriged_results.B_a_adult_prediction_variance ) )
+            ) )
+        )
+        
+        ### Add biomass values to all ages for kriged biomass results
+        kriged_results_output = (
+            kriged_results
+            .merge( self.biology[ 'weight' ][ 'proportions' ][ 'sex_age_weight_proportions_df' ] , 
+                    on = [ 'stratum_num' , 'sex' ] )
+            .assign( biomass_sex_age = lambda x: x.weight_sex_proportion_all * x.biomass_total )
+            .loc[ : , [ 'centroid_latitude' , 'centroid_longitude' , 'stratum_num' , 
+                        'cell_area_nmi2' , 'sex' , 'age' , 'biomass_unaged' , 'biomass_aged' ,
+                        'biomass_total' , 'biomass_adult_cell_CV' , 'biomass_sex_age' ] ]
+        )
+        
+        ### Assign results to an attribute
+        self.statistics[ 'kriging' ].update(
+            {
+                'apportioned_kriged_biomass_df': kriged_results_output ,
+            }
+        )
```

Comment:
I was curious about if there's any performance difference between `.update` or direct assignment. It looks like that's not the case, just that with `.update` one could assign many keys at a time.